### ISSUE TITLE
TypeError: ["'", 'e', ':', '?'] has type list, but expected one of: bytes, unicode

### ISSUE THREAD

**Issue by alexander-winkler** [OCR4all/OCR4all#10]
Hello! I'm using calamari from within the new OCR4all tool under Linux Mint 19.1 Tessa. The OCR process stops with the following error:

```
Traceback (most recent call last):
  File "/usr/local/bin/calamari-predict", line 11, in 
    load_entry_point('calamari-ocr==0.3.1', 'console_scripts', 'calamari-predict')()
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.1-py3.6.egg/calamari_ocr/scripts/predict.py", line 151, in main
    run(args)
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.1-py3.6.egg/calamari_ocr/scripts/predict.py", line 74, in run
    prediction = voter.vote_prediction_result(result)
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.1-py3.6.egg/calamari_ocr/ocr/voting/voter.py", line 19, in vote_prediction_result
    return self.vote_prediction_result_tuple(tuple(prediction_results))
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.1-py3.6.egg/calamari_ocr/ocr/voting/voter.py", line 45, in vote_prediction_result_tuple
    p.sentence = [c for c, _ in sv.process_text(sentences)]
TypeError: ["'", 'e', ':', '?'] has type list, but expected one of: bytes, unicode
```

What can I do in order to avoid the TypeError?

Many thanks!

**Comment by NesbiDevelopment** [comment]
Hello! This TypeError is new to us.
I'll gladly help but I do need more information.

First of all what version/tag of OCR4all do you use?
e.g.
``` bash
docker pull ls6uniwue/ocr4all                #uses latest/stable
docker pull ls6uniwue/ocr4all:latest      #uses latest/stable
docker pull ls6uniwue/ocr4all:dev         #uses dev
docker pull ls6uniwue/ocr4all:nightly    #uses nightly
```
and building here from source master uses nightly.

Could you tell me more about your training?
* What settings did you use? (Screenshot would be helpful)
* Did you solely train your model in OCR4all and could you provide the model? (Providing the model makes it easier to reproduce the error)

**Comment by alexander-winkler** [comment]
Hello!
I have just pulled the latest version via the commands you have indicated and I was able to reproduce the error with the following settings:
![image](https://user-images.githubusercontent.com/52370148/60806357-0d73ee80-a183-11e9-9576-ffe64394d444.png)

Using only on of the two default models, there is no error. I think that there has occasionally been the same error also using own models. I should add that the error always occurs on a specific page. It would be useful if the error didn't cause the Recognition to stop.

**Comment by NesbiDevelopment** [comment]
Thank you for this info.
This helps us to narrow down the error to a possible defect recognition/ground truth file.
Could you be so kind to provide us with these recognition and ground truth files?

They would be in either:
Predictions: ```<your-ocr4all-folder>/data/<project-name>/processing/<page-nr>/<region>/<line>.pred.txt```
Ground Truths: ```<your-ocr4all-folder>/data/<project-name>/processing/<page-nr>/<region>/<line>.pred.txt```

Or if ```<your-ocr4all-folder>/data/<project-name>/processing/<page-nr>/``` does not exist in:

Ground Truths/Predictions: ```<your-ocr4all-folder>/data/<project-name>/processing/<page-nr>.xml```

We sadly can't do much without those files, since we can't reproduce the error.

We will then of cause work together with the folks of calamari to fix it. The Recognition should then at least not stop on the problem.

**Comment by alexander-winkler** [comment]
According to the following console output, I suppose the relevant files are those in the attached tar-archive.
![image](https://user-images.githubusercontent.com/52370148/60810668-e4a52680-a18d-11e9-918d-c749b95b403c.png)

[ocr4all_TypeError.tar.gz](https://github.com/OCR4all/OCR4all/files/3368354/ocr4all_TypeError.tar.gz)

**Comment by NesbiDevelopment** [comment]
Thank you very much, I was able to reproduce the error:
```
2019-07-08 13:16:43.650847: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
Traceback (most recent call last):
  File "/usr/local/bin/calamari-predict", line 11, in 
    load_entry_point('calamari-ocr==0.3.3', 'console_scripts', 'calamari-predict')()
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.3-py3.6.egg/calamari_ocr/scripts/predict.py", line 157, in main
    run(args)
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.3-py3.6.egg/calamari_ocr/scripts/predict.py", line 76, in run
    prediction = voter.vote_prediction_result(result)
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.3-py3.6.egg/calamari_ocr/ocr/voting/voter.py", line 19, in vote_prediction_result
    return self.vote_prediction_result_tuple(tuple(prediction_results))
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.3-py3.6.egg/calamari_ocr/ocr/voting/voter.py", line 45, in vote_prediction_result_tuple
    p.sentence = [c for c, _ in sv.process_text(sentences)]
TypeError: ["'", 'e', ':', '?'] has type list, but expected one of: bytes, unicode
Exception ignored in: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.3-py3.6.egg/calamari_ocr/ocr/predictor.py", line 278, in predict_dataset
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.3-py3.6.egg/calamari_ocr/ocr/datasets/input_dataset.py", line 301, in __exit__
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.3-py3.6.egg/calamari_ocr/ocr/datasets/input_dataset.py", line 148, in __exit__
  File "/usr/lib/python3.6/logging/__init__.py", line 1295, in debug
  File "/usr/lib/python3.6/logging/__init__.py", line 1548, in isEnabledFor
TypeError: '>=' not supported between instances of 'int' and 'NoneType'
Exception ignored in: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.3-py3.6.egg/calamari_ocr/ocr/backends/tensorflow_backend/tensorflow_model.py", line 528, in predict
AttributeError: 'NoneType' object has no attribute 'OutOfRangeError'
Exception ignored in: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/calamari_ocr-0.3.3-py3.6.egg/calamari_ocr/ocr/backends/tensorflow_backend/tensorflow_model.py", line 528, in predict
AttributeError: 'NoneType' object has no attribute 'OutOfRangeError'
```

I will now investigate further.

**Comment by NesbiDevelopment** [comment]
Hello alexander-winkler, we investigated the problem further with the Calamari-OCR group.
The problem seemed to arise with the usage of multiple models with some incompatibilities.
Those incompatibilities have been fixed in Calamari and the newest Version of Calamari will shortly be added to OCR4all.

In the meantime, mixing some models can on some circumstances result in that error.
Using different or removing some of the models from the selection may help to avoid this error.
